### PR TITLE
add test cases for Enumerable methods

### PR DESCRIPTION
**Pull request by stratospherique**
I used Rspec to test 6 of the self implemented Enumerable methods. The test varies from result comparison (i.e. array matching, values) to errors expecting

### CODE REVIEW

File: Enumerable-Methods/spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    end
+  end
+
+  describe "#my_all" do 
+    it "checks if all the elements are complient with a condition" do
+      expect([1,8,9,2,10].my_all{|x| x>0}).to be(true)
+    end
+
+    it "return an error if the passed array is not flattened" do
+      expect{[1,[8,9],2,10].my_all{|x| x>0}}.to raise_error
```
Comment (inhaq) [inline Enumerable-Methods/spec/spec_helper.rb]: you can use the `raise_error` matcher to specify a block of code raises an error but here there is no specific error given to it

**Review by inhaq — CHANGES_REQUESTED** [review]
Kindly update minor issue, once fixed kindly resubmit for review.

File: Enumerable-Methods/spec/spec_helper.rb (hunk 1 above)
Comment (stratospherique) [reply Enumerable-Methods/spec/spec_helper.rb]: `[1,[8,9],2,10].all?{|x| x>0}`
Traceback (most recent call last):
        6: from modules.rb:195:in `<main>'
        5: from modules.rb:45:in `my_all'
        4: from modules.rb:14:in `my_each'
        3: from modules.rb:14:in `each'
        2: from modules.rb:15:in `block in my_each'
        1: from modules.rb:46:in `block in my_all'
modules.rb:195:in `block in <main>': undefined method `>' for [8, 9]:Array (NoMethodError)

using all? on a non flattened array will raise a NoMethodError

**Review by inhaq — APPROVED** [review]
It's 👍🏻, go ahead and submit the project completion form.
Good luck with the next projects.

**Comment by stratospherique** [comment]
Thanks ;)

**Pull request merged**